test/e2e-tests.sh

```diff
@@ -66,9 +66,11 @@ kubectl delete buildtemplates --all
 # Run the tests
 
 header "Running tests"
-
 ko apply -R -f test/ || fail_test
 
+header "Running Go e2e tests"
+GOCACHE=off go test -tags e2e ./test/e2e/... || fail_test
```

Comment:
Use report_go_test here instead of `go test` so you'll get individual results on testgrid and in the gubernator report.

Also, don't `fail_test` here. Fail after checking the results, so you'll get all results of the test. Better run the go tests first, since you're not checking the results here (even better: move everything to a function). Something like:

```
local failed=0
report_go_test ... || failed=1
run_old_ugly_tests_in_shell_script || failed=1
ko apply ... || failed=1
(( failed )) && abort_test
success
```